CHANGELOG.md

```diff
@@ -52,6 +52,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * The `empty` condition in templates is true when the commit makes no change to
   the three compared to its parents.
 
+* Setting the new `ui.unique-prefixes` option to `underscore` makes `jj log`
+  highlight the shortest unique prefix of every commit and change id.
```

Comment:
> For now, `none` is the default, since the `underscore` value messes up copy and pasting of ids.

How about making it a colon for now then? Both iTerm2 and Gnome Terminal stop double-click selection at colon.

Reply:
Colon is a good idea -- it does seem to separate words and looks pretty good. Do you think the default should be "none" or "colon"?

Reply:
My general feeling is that we're still early enough in the project that we should experiment quite freely before it's too late :) So my vote is for "colon".

Reply:
+1 for colon instead of none as the default (it's a nice feature that should be on by default IMO), re: copy paste, I have to use PuTTY for work and even that stops at colon, so that's good -- one way to side step this would be of course to use color or something like that for differentiation (grey out the non unique part of the id maybe).

Reply:
Just fyi. I've seen some people get confused with `49904:f3e95e5a5895` (`{rev}:{node|short}`) in Mercurial because it's also valid as a range expression.

Reply:
Oh, that's a good point! iTerm2 seems to default to "/-+\~_." as part of a word (in addition to letters, presumably). Gnome terminal uses a larger set, but `$`, `^`, and `*` all seem to work in both of those two. I don't really have a preference among them. Anyone else?

Reply:
I would pick `$` since it's unlikely we would add infix `$` op (and it looks like a mangled symbol), but it might be hard to distinguish from `8`.

FWIW, foot uses a different set of characters. ,│`|:"'()[]{}<> and whitespaces are "delimiters".
https://codeberg.org/dnkl/foot/src/branch/master/doc/foot.ini.5.scd#L306

Reply:
How about `,`? `^` looks weird to me, but not terrible.

I never double-click to select, so I'm not sure how important that is. For selecting the usual way, all symbols are equally not great. Once we get to `$`, I'd personally prefer `_`. `$` looks a bit distracting to me. 

If people prefer it, i can live with it, especially if it's a default that can be changed.  I could implement two or three possible symbols (`$`,`:`,`_`), or make it possible to use any symbol (this might be overkill).

Finally, we could implement underline + bold highlighting, perhaps pretty soon. It doesn't interfere with copy/paste at all.

Reply:
I don't care much about which one we pick. I don't think it's worth making it configurable since it's supposed to just be a temporary thing.

An alternative to underling and/or bold, I like @avamsi's proposal of graying out the remainder of the string. I'm not sure I'll like it better, but let's keep it in mind.

Reply:
I guess , will make it look like there are several ids. I'll add `.` for consideration instead.

Greying out sounds nice, we should certainly consider it.

Reply:
OK, currently `.` is my favorite. 

![image](https://user-images.githubusercontent.com/4123047/212793471-50711c02-f1fa-4774-a53b-7001d705f6ce.png)


One more option is brackets. I'm not sure which is better:

Bracket A
![image](https://user-images.githubusercontent.com/4123047/212793426-8827ced6-f05f-426e-b2c8-a7f32abf0feb.png)

Bracket B
![image](https://user-images.githubusercontent.com/4123047/212793552-fe577336-4c35-4800-8fec-1c8239516007.png)


If I don't hear otherwise, or think of some problem with it myself, I'm going with `.` for now.

Reply:
Bracket B looks good to me. It matches the common expectation that the part within brackets is optional.

Reply:
Bracket B it is then. I also made it the default since that seemed to be your preference, but I don't feel strongly about that one way or the other.

Reply:
I also don't feel strongly. By making it the default, we're more likely to get feedback. On the other hand, it might be better to not change it now and then change it again soon if you think you're going to get around to styling it differently soon (with colors or underlining). Up to you.

Reply:
🤷